### ISSUE TITLE
a question

### ISSUE THREAD

**Issue by xylovezxy** [beichenzbc/Long-CLIP#75]
您好，我之前写了一段代码，在一份多模态二分类数据集上训练clip去进行预测，我用这句代码区加载模型：

clip_model = CLIPModel.from_pretrained("/hy-tmp/clip-vit-base-patch32/")
processor = CLIPProcessor.from_pretrained("/hy-tmp/clip-vit-base-patch32/")
class MovieDataset(Dataset):
    def __init__(self, dataframe, processor):
        self.dataframe = dataframe
        self.processor = processor

    def __len__(self):
        return len(self.dataframe)

    def __getitem__(self, idx):
        image_path = self.dataframe.iloc[idx]['img_path']
        title = self.dataframe.iloc[idx]['电影简介']
        target = self.dataframe.iloc[idx]['target']

        image = Image.open(image_path).convert('RGB')
        inputs = self.processor(text=title, images=image, return_tensors="pt", padding=True)
        inputs['label'] = torch.tensor(target, dtype=torch.float)

        # squeeze to remove unnecessary batch dimension
        inputs = {k: v.squeeze(0) for k, v in inputs.items()}
        
        return inputs
    
train_dataset = MovieDataset(df_train, processor)
val_dataset = MovieDataset(df_test, processor)

from torch.nn.utils.rnn import pad_sequence

def collate_fn(batch):
    input_ids = [item['input_ids'] for item in batch]
    pixel_values = torch.stack([item['pixel_values'] for item in batch])
    attention_mask = [item['attention_mask'] for item in batch]
    labels = torch.tensor([item['label'] for item in batch])

    # 对input_ids和attention_mask进行填充
    input_ids_padded = pad_sequence(input_ids, batch_first=True, padding_value=processor.tokenizer.pad_token_id)
    attention_mask_padded = pad_sequence(attention_mask, batch_first=True, padding_value=0)

    return {
        'input_ids': input_ids_padded,
        'pixel_values': pixel_values,
        'attention_mask': attention_mask_padded,
        'label': labels
    }

train_loader = DataLoader(train_dataset, batch_size=16, shuffle=True, collate_fn=collate_fn)
val_loader = DataLoader(val_dataset, batch_size=16, collate_fn=collate_fn)

但是现在我想尝试使用longclip，您提供的模型只有一个pt文件，我尝试了使用longclip.load但是无法正确运行，请问应该如何解决呢？

**Comment by beichenzbc** [comment]
您可以先把这个仓库git clone到您当前的文件夹下面，再把我们的checkpoint从huggingface里面下载下来，这样就可以只用longclip.load(xxx<您本地.pt的路径>/longclip-B.pt)进行加载

**Comment by xylovezxy** [comment]
这样的方法我尝试过了，但是无法成功运行我的代码，好像是longclip与clip的结构有所不同？主要是因为
clip_model = CLIPModel.from_pretrained("/hy-tmp/clip-vit-base-patch32/")
processor = CLIPProcessor.from_pretrained("/hy-tmp/clip-vit-base-patch32/")
和
longclip.load(xxx<您本地.pt的路径>/longclip-B.pt)
获得的model和processor不一样，导致我无法续用之前的代码

**Comment by beichenzbc** [comment]
您使用的是huggingface代码格式，这个和我们github上pytorch版本的代码格式不太一致，一般不能通用。这个改起来比较容易，您可以自行修改下代码。

**Comment by xylovezxy** [comment]
请问是把我的代码改成适配pytorch版本的格式吗

**Comment by beichenzbc** [comment]
对的，你的其他代码需要修改成pytorch版本的

**Comment by eternal8080** [comment]
请问您解决这个问题了嘛，我最近也是想直接把clip换为Longclip，这样做可行吗？longclip和clip在模型架构上没有区别吧

**Comment by xylovezxy** [comment]
> 请问您解决这个问题了嘛，我最近也是想直接把clip换为Longclip，这样做可行吗？longclip和clip在模型架构上没有区别吧

你好,是可行的，但是需要把用clip的代码改一下
